onnx/defs/shape_inference.h

```diff
@@ -393,25 +393,35 @@ inline void appendSingleDimCopiedFromInputTypeToOutputType(
   *dim = input_type->tensor_type().shape().dim(static_cast<int>(fromDimIndex));
 }
 
+inline void propagateShape(const TypeProto* from_type, TypeProto* to_type) {
+  if (TypeProto::kTensorType == from_type->value_case() &&
+      TypeProto::kTensorType == to_type->value_case()) {
+    // If input shape is "uknown", the corresponding should be "unknown" too.
+    // The way to make output shape unknown is not to assign it any value.
+    if (hasShape(*from_type)) {
+      *to_type->mutable_tensor_type()->mutable_shape() =
+          from_type->tensor_type().shape();
+    }
+  }
+}
+
 inline void propagateShapeFromInputToOutput(
     InferenceContext& ctx,
     size_t inputIndex,
     size_t outputIndex) {
   auto output_type = ctx.getOutputType(outputIndex);
   auto input_type = ctx.getInputType(inputIndex);
 
-  if (TypeProto::kTensorType != input_type->value_case() ||
-      TypeProto::kTensorType != output_type->value_case()) {
+  if (TypeProto::kTensorType == input_type->value_case() &&
```

Comment:
It would be more general to move this if-then-else logic into "propagateShape". For example, then it would work for sequence(sequence(tensor(...))) as well.

Reply:
@BowenBao : does the above make sense? Thanks!

Reply:
@gramalingam sorry for the late reply, will take a look and make updates accordingly.

Reply:
updated